### ISSUE TITLE
Todo Tutorial Feedback -- Issues, Comments, Challenges

### ISSUE THREAD

**Issue by virkt25** [strongloop/loopback-next#1206]
## Description / Steps to reproduce / Feature proposal

As part of #729 I sat down with a user with no prior LoopBack experience (but a developer) and had him go through what is currently known as `example-getting-started` (pending rename to `example-todo`) Tutorial. I made notes during this process and gathered the following feedback from the user. 

---

- Node 9 CLI issues after global install
- Why do I need to create a class?
- What is loopbackBuild?
- Looking for views … coming from react background … 
    - Explain later
- Missed the Next Step … Make Next Step a heading instead of Navigation
- Renamed `index.ts` to `model.ts` since all folders have `index.ts` in them
- Skipped legacy juggler bridge step since he doesn’t know what that is and said “I couldn’t care less”
- Added export to index.ts first but that led to a warning in VSCode that module doesn’t exist.
- Under Model section … what is Repository? Entity? Etc? 
- For Datasource section, expected `config` directory to exist already. 
    - Saw instructions for creating the `config` folder. 
    - Questioned Why?
    - Doesn’t know what juggler is … 
    - Dislikes the config/datasources.json folder
    - Why wasn’t the config folder created?
    - Thought the config file was good enough — not sure of the purpose of the ts file. 
- Repositories folder
    - What do I put in `index.ts` … also created it as `repositories.ts`
    - Export * from ‘pwd’ ?? … 
- Doesn’t mention to go to /swagger-ui
- Redirect to loopback.io isn’t intuitive (link to API Explorer in Docs)
- Missing id field in GET/todo/id (so id=undefined) resulted in a 500 but should’ve been 4XX (Bad request)
- Id shows up in all “try it out” options as 0 but it is ignored in various calls including Create
- Can you patch the ID?
- Response for PATCH / PUT is `true` instead of JSON as one would expect
- Was able to delete id=2 when it doesn’t exist. Returned a success message.
- Feels like an architecture that scales well but a lot of new concepts

- Be nice to add the tutorial to [todomvc](https://github.com/tastejs/todomvc/tree/master/examples)

cc: @strongloop/lb-next-dev I think we should work on addressing some of these areas of concerns with the next release (after DP2).

**Comment by shimks** [comment]
This is some good feedback 👌

**Comment by shimks** [comment]
@virkt25 Could you update this task with an appropriate acceptance criteria? At first glance, this issue seems like it'll pan out to be a spike to see how the feedback could be addressed, but what do you think?

**Comment by dhmlau** [comment]
I agree that we need to work things out to have a set of acceptance criteria.  However, I don't think there's a need for a spike, rather than fixing things up.

**Comment by virkt25** [comment]
No Spikes. I'll take an attempt at creating a Todo List (get it) for the tasks with some criteria.

**Comment by fjanon** [comment]
Adding some issues in the LB4 todo tutorial

https://loopback.io/doc/en/lb4/todo-tutorial-repository.html

1) I think the code immediately following:
NOTE: The index.ts file is an export helper file
should be part of the note. It's confusing here, we are looking for the todo code for index.ts that's missing.
2) the code for the index.ts is missing

In the controller page and code:
https://loopback.io/doc/en/lb4/todo-tutorial-controller.html

The return type for the method is Promise<boolean>:

@put('/todos/{id}')
  async replaceTodo(
    @param.path.number('id') id: number,
    @requestBody() todo: Todo,
    // Must change to Promise<void>
  ): Promise<boolean> {
    return await this.todoRepo.replaceById(id, todo);
  }

gives an error at compile and runtime times.

The correct return type for the method is Promise<void>:

  // from the finished example
  async replaceTodo(
    @param.path.number('id') id: number,
    @requestBody() todo: Todo,
  ): Promise<void> {
    await this.todoRepo.replaceById(id, todo);
  }

**Comment by bajtos** [comment]
@fjanon thank you for the feedback! Would you mind contributing the proposed changes yourself? See [DEVELOPING.md](https://github.com/strongloop/loopback-next/blob/master/docs/site/DEVELOPING.md) and [contrib guide](https://loopback.io/doc/en/contrib/index.html) to get started.

**Comment by dhmlau** [comment]
@fjanon, I believe I've fixed them via PR https://github.com/strongloop/loopback-next/pull/1846.  
For creating controller in the tutorial, we now ask our users to select the `REST Controller with CRUD functions` option so that most of the code are generated.  Thanks.
